### DISCUSSION TITLE
How can I learn more about how DVC stores the files in an object storage?

### DISCUSSION THREAD

**Discussion by FeryET** [treeverse/dvc#7923] [General]
Hello.

I'm trying to integrate a labelling system with our data registry (which is a single object storage atm), but I also want to use the same registry with dvc too, so that our raw data reside only in one object storage, which is accessible by both tools.

How can I read more about how data is actually stored in a bucket via dvc? Is it unpacked in a flat way and then organized by a relational db?

Thanks.

**Comment by pmrowla** [comment]
DVC stores files as content-addressable storage according to each file's hash. It does not use any relational DB.

 https://dvc.org/doc/user-guide/project-structure/internal-files#structure-of-the-cache-directory

**Reply by FeryET** [reply]
Is it the same in the remote object storage?

**Reply by pmrowla** [reply]
Yes, the structure of the local cache and remote is the same
